### PR TITLE
Add tests post market page

### PR DESCRIPTION
**Pull request by tbille**
# Summary

- Add tests for POST endpoint
- Seperated in 2 files:
  - One for fields updates
  - One for screenshots/icon uploads

# QA

- `./run test`

### CODE REVIEW

File: tests/tests_post_market.py [hunk 1]
```diff
@@ -0,0 +1,568 @@
...
+        called = responses.calls[1]
+        self.assertEqual(
+            'https://login.ubuntu.com/api/v2/tokens/refresh',
+            called.request.url)
+
+        assert response.status_code == 302
+        assert response.location == self._get_redirect()
+
+    @responses.activate
+    def test_update_valid_fied(self):
```
Comment (bartaz) [inline tests/tests_post_market.py]: Typo in function name `fied`

File: tests/tests_post_market.py [hunk 2]
```diff
@@ -0,0 +1,568 @@
...
+    def test_post_no_data(self):
+        response = self.client.post(
+            self.endpoint_url,
+        )
+
+        assert response.status_code == 302
+        assert response.location == self._get_location()
+
+    @responses.activate
+    def test_update_invalid_fied(self):
```
Comment (bartaz) [inline tests/tests_post_market.py]: Another typo: `fied`

File: tests/tests_post_binary_metadata.py [hunk 3]
```diff
@@ -0,0 +1,440 @@
+import unittest
+import responses
+import json
+import io
+from tests.endpoint_testing import BaseTestCases
+
+
+class PostBinaryMetadataMarketPage(BaseTestCases.BaseAppTesting):
```
Comment (bartaz) [inline tests/tests_post_binary_metadata.py]: There are no tests for removing screenshots?
We could also have a test for removing icon (even if we don't have UI for it yet).

I'm not sure if I seen a test for when images fail to upload (icon is wrong size or screenshots in wrong format).

Also, I think we still have a bug that if you add twice same image (which is possible in our UI) upload will fail. Maybe we could test this specific edge case (and fix it). But not sure if it needs to be part of this PR if it's too big.

**Comment by nottrobin** [comment]
@bartaz can you add issues for the missing cases?
